prow/types.go

```diff
@@ -26,11 +26,11 @@ type JobSpec struct {
 }
 
 type JobStatus struct {
-	State          string      `json:"state"`
-	StartTime      metav1.Time `json:"startTime"`
-	CompletionTime metav1.Time `json:"completionTime"`
-	URL            string      `json:"url"`
-	BuildID        string      `json:"build_id"`
+	State          string       `json:"state"`
+	StartTime      metav1.Time  `json:"startTime"`
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
```

Comment:
Why would we do this?  That’s not the schema.

Reply:
Every job should have a completion time.  Some may not have started times, but we might be able to add that to the index.

Reply:
pending jobs will not have a completion time.

Reply:
and topic post on this PR links to the test-infra type definition, showing that it is the schema over there ;).

Reply:
Ah, this is the one place that actually uses non completed jobs.  Hrm... I really don't want to deal with the ripples in the code base on this.  Use time.IsZero()

Reply:
Note that right now you have then problem that start time is zero.  So we either need to fake start time from the index.  But all the code was just fine with zero start time and omit empty, so not sure why you need then pointer.